### PR TITLE
[azmq] Update to latest commit

### PR DESCRIPTION
**Pull request by NancyLi1013**
Update `azmq `to latest commit and support `boost `1.70.

Related issue #10277 

Note: No feature needs to test.

### CODE REVIEW

**Review by LilyWangL — APPROVED** [review]

File: ports/azmq/CONTROL [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Source: azmq
-Version: 1.0.2
+Version: 2020-03-03
```
Comment (strega-nil) [inline ports/azmq/CONTROL]: This should probably be `2019-05-01`, no?

Comment (NancyLi1013) [reply ports/azmq/CONTROL]: @strega-nil Thanks for your review. In this case, we should use the date that the commit was accessed by us, namely the date that we updated the source version.
Plese refer to this [control-files.md](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md).

Comment (strega-nil) [reply ports/azmq/CONTROL]: @NancyLi1013 oh, okay, cool! Thanks :)

**Comment by strega-nil** [comment]
Cool, thanks!

**Pull request merged**